### ISSUE TITLE
ARIA labels on main area widget are incorrect (always referring to a notebook)

### ISSUE THREAD

**Issue by krassowski** [jupyterlab/jupyterlab#13045]
## Description

https://github.com/jupyterlab/jupyterlab/blob/b57ec962a7b8cda65bb89a72223f195564f8387a/packages/apputils/src/mainareawidget.ts#L46-L50

says "notebook content" when using a file editor or opening a CSV file, etc, but it should not.

We could:
- change it to "main area content" and "main area toolbar"
- in each of the `MainAreaWidget` subclasses set a proper text
  - we could take the string from a getter so subclasses have less work (getter so that we can translate)
  - if we want to be more strict we could use an abstract getter and force all subclasses of `MainAreaWidget` to provide an aria-label since 4.0

**Comment by thevyomonaut** [comment]
I would like to work on this, please! I am a first timer, so some help would be appreciated.

**Comment by scmmmh** [comment]
I've been looking into this and I have a question, which may be caused by my misunderstanding of the architecture. Since the MainAreaWidget is generic, shouldn't the aria-label value be taken from the Widget that is used to parameterise the MainAreaWidget?

**Comment by krassowski** [comment]
@scmmmh you are right, sorry about inaccurate suggestion. Indeed for the most part we do not subclass `MainAreaWidget` (`DocumentWidget` aside). Using a property from the parameter Widget seems like a good idea!

Maybe it is the Widget's responsibility to set the `aria-label` and `rule` on its node, and not a responsibility of `MainAreaWidget`, so we could remove `MainAreaWidget` from the equation if we make changes in lumino's Widget instead?

**Comment by scmmmh** [comment]
@krassowski I agree, the Widget is probably the better place to add that support. I'll look into that and then this issue can be used to track updating the MainAreaWidget when the lumino Widget has been updated.

**Comment by m158261** [comment]
There is a test that covers the incorrect behavior described in this issue. This test will need to be updated when this issue is worked on. The test can be found in this merged PR. https://github.com/jupyterlab/jupyterlab/pull/15137

**Comment by dnlzrgz** [comment]
Hi everyone,

I’ve been looking into this issue and noticed that there are currently no open PR addressing it. I’ve checked the previous PR and reviewed the test mentioned, but I wanted to confirm if this issue is still relevant before I create a pull request of my own.

**Comment by krassowski** [comment]
Yes, this issue is still relevant - feel welcome to open a PR!
